scripts/generate.py

```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/env python
+
+"""Generate build and run scripts for Unikraft application.
+
+Use KrafKit and Make for building. Use QEMU, Firecracker and Krafkit
+for running.
+"""
+
+import sys
+import os
+import stat
+import yaml
+
+TEMPLATE_BUILD_MAKE = """#!/bin/sh
+
+make distclean
+UK_DEFCONFIG=$(pwd)/defconfigs/{} make defconfig
+make prepare
+make -j $(nproc)
+"""
+
+TEMPLATE_BUILD_KRAFT = """#!/bin/sh
+
+rm -fr .unikraft
+rm -f .config.*
+kraft build --log-level debug --log-type basic --target {} --plat {}
+"""
+
+TEMPLATE_RUN_QEMU_HEADER = """#!/bin/sh
+
+kernel="{}"
+cmd="{}"
+
+if test $# -eq 1; then
+    kernel="$1"
+fi
+"""
+
+RUN_COMMON_NET_COMMANDS = """
+# Remove previously created network interfaces.
+sudo ip link set dev tap0 down 2> /dev/null
+sudo ip link del dev tap0 2> /dev/null
+sudo ip link set dev virbr0 down 2> /dev/null
+sudo ip link del dev virbr0 2> /dev/null
+"""
+
+RUN_QEMU_NET_COMMANDS = """
+# Create bridge interface for QEMU networking.
+sudo ip link add dev virbr0 type bridge
+sudo ip address add 172.44.0.1/24 dev virbr0
+sudo ip link set dev virbr0 up
+"""
+
+RUN_FIRECRACKER_NET_COMMANDS = """
+# Create tap interface for Firecracker networking.
+sudo ip tuntap add dev tap0 mode tap
+sudo ip address add 172.44.0.1/24 dev tap0
+sudo ip link set dev tap0 up
+"""
+
+RUN_KRAFT_NET_COMMANDS = """
+# Create bridge interface for KraftKit networking.
+sudo kraft net create -n 172.44.0.1/24 virbr0
+"""
+
+TEMPLATE_RUN_FIRECRACKER_HEADER = """#!/bin/sh
+
+config="{}"
+
+if test $# -eq 1; then
+    config="$1"
+fi
+"""
+
+RUN_FIRECRACKER_TRAILER = """
+# Remove previously created files.
+sudo rm -f /tmp/firecracker.log
+> /tmp/firecracker.log
+sudo rm -f /tmp/firecracker.socket
+
+sudo firecracker-x86_64 \\
```

Comment:
```suggestion
firecracker-x86_64 \\
```
Does firecracker need root? It worked fine for me without this.